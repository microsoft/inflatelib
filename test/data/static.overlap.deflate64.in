
# Tests several scenarios where window copies overlap itself. The inputs are as follows
#   * Write 1344 characters worth of "London Bridge"
#   * Repeat the string 47 times: (63168, 1344)
#   * Repeat the string 48 times offset the length of one string: (64512, 1344)
#   * Repeat the string 48 times offset the length of 48 strings: (64512, 64512)
#   * Write the literal '0'
#   * Copy '0' 1023 times: (1023, 1)
#   * Repeat the string 48 times offset at the start of the buffer: (64512, 65536)
# NOTE: This test is only applicable to Deflate64
>1
1       # bfinal = true
01      # Compressed with static codes

# Encoded Data:
>>1
01111100 10011111 10011110 10010100 10011111 10011110 01010000 01110010 10100010 10011001 10010100 10010111 10010101 01010000 10011001 10100011 01010000 10010110 10010001 10011100 10011100 10011001 10011110 10010111 01010000 10010100 10011111 10100111 10011110 01011100 00111010
01110110 10010001 10011100 10011100 10011001 10011110 10010111 01010000 10010100 10011111 10100111 10011110 01011100 01010000 10010110 10010001 10011100 10011100 10011001 10011110 10010111 01010000 10010100 10011111 10100111 10011110 01011100 00111010
01111100 10011111 10011110 10010100 10011111 10011110 01010000 01110010 10100010 10011001 10010100 10010111 10010101 01010000 10011001 10100011 01010000 10010110 10010001 10011100 10011100 10011001 10011110 10010111 01010000 10010100 10011111 10100111 10011110 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
01110010 10100101 10011001 10011100 10010100 01010000 10011001 10100100 01010000 10100101 10100000 01010000 10100111 10011001 10100100 10011000 01010000 10100111 10011111 10011111 10010100 01010000 10010001 10011110 10010100 01010000 10010011 10011100 10010001 10101001 01011100 00111010
10000111 10011111 10011111 10010100 01010000 10010001 10011110 10010100 01010000 10010011 10011100 10010001 10101001 01011100 01010000 10100111 10011111 10011111 10010100 01010000 10010001 10011110 10010100 01010000 10010011 10011100 10010001 10101001 01011100 00111010
01110010 10100101 10011001 10011100 10010100 01010000 10011001 10100100 01010000 10100101 10100000 01010000 10100111 10011001 10100100 10011000 01010000 10100111 10011111 10011111 10010100 01010000 10010001 10011110 10010100 01010000 10010011 10011100 10010001 10101001 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
10000111 10011111 10011111 10010100 01010000 10010001 10011110 10010100 01010000 10010011 10011100 10010001 10101001 01010000 10100111 10011001 10011100 10011100 01010000 10100111 10010001 10100011 10011000 01010000 10010001 10100111 10010001 10101001 01011100 00111010
10000111 10010001 10100011 10011000 01010000 10010001 10100111 10010001 10101001 01011100 01010000 10100111 10010001 10100011 10011000 01010000 10010001 10100111 10010001 10101001 01011100 00111010
10000111 10011111 10011111 10010100 01010000 10010001 10011110 10010100 01010000 10010011 10011100 10010001 10101001 01010000 10100111 10011001 10011100 10011100 01010000 10100111 10010001 10100011 10011000 01010000 10010001 10100111 10010001 10101001 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
01110010 10100101 10011001 10011100 10010100 01010000 10011001 10100100 01010000 10100101 10100000 01010000 10100111 10011001 10100100 10011000 01010000 10010010 10100010 10011001 10010011 10011011 10100011 01010000 10010001 10011110 10010100 01010000 10011101 10011111 10100010 10100100 10010001 10100010 01011100 00111010
01110010 10100010 10011001 10010011 10011011 10100011 01010000 10010001 10011110 10010100 01010000 10011101 10011111 10100010 10100100 10010001 10100010 01011100 01010000 10010010 10100010 10011001 10010011 10011011 10100011 01010000 10010001 10011110 10010100 01010000 10011101 10011111 10100010 10100100 10010001 10100010 01011100 00111010
01110010 10100101 10011001 10011100 10010100 01010000 10011001 10100100 01010000 10100101 10100000 01010000 10100111 10011001 10100100 10011000 01010000 10010010 10100010 10011001 10010011 10011011 10100011 01010000 10010001 10011110 10010100 01010000 10011101 10011111 10100010 10100100 10010001 10100010 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
01110010 10100010 10011001 10010011 10011011 10100011 01010000 10010001 10011110 10010100 01010000 10011101 10011111 10100010 10100100 10010001 10100010 01010000 10100111 10011001 10011100 10011100 01010000 10011110 10011111 10100100 01010000 10100011 10100100 10010001 10101001 01011100 00111010
10000111 10011001 10011100 10011100 01010000 10011110 10011111 10100100 01010000 10100011 10100100 10010001 10101001 01011100 01010000 10100111 10011001 10011100 10011100 01010000 10011110 10011111 10100100 01010000 10100011 10100100 10010001 10101001 01011100 00111010
01110010 10100010 10011001 10010011 10011011 10100011 01010000 10010001 10011110 10010100 01010000 10011101 10011111 10100010 10100100 10010001 10100010 01010000 10100111 10011001 10011100 10011100 01010000 10011110 10011111 10100100 01010000 10100011 10100100 10010001 10101001 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
01110010 10100101 10011001 10011100 10010100 01010000 10011001 10100100 01010000 10100101 10100000 01010000 10100111 10011001 10100100 10011000 01010000 10011001 10100010 10011111 10011110 01010000 10010001 10011110 10010100 01010000 10100011 10100100 10010101 10010101 10011100 01011100 00111010
01111001 10100010 10011111 10011110 01010000 10010001 10011110 10010100 01010000 10100011 10100100 10010101 10010101 10011100 01011100 01010000 10011001 10100010 10011111 10011110 01010000 10010001 10011110 10010100 01010000 10100011 10100100 10010101 10010101 10011100 01011100 00111010
01110010 10100101 10011001 10011100 10010100 01010000 10011001 10100100 01010000 10100101 10100000 01010000 10100111 10011001 10100100 10011000 01010000 10011001 10100010 10011111 10011110 01010000 10010001 10011110 10010100 01010000 10100011 10100100 10010101 10010101 10011100 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
01111001 10100010 10011111 10011110 01010000 10010001 10011110 10010100 01010000 10100011 10100100 10010101 10010101 10011100 01010000 10100111 10011001 10011100 10011100 01010000 10010010 10010101 10011110 10010100 01010000 10010001 10011110 10010100 01010000 10010010 10011111 10100111 01011100 00111010
01110010 10010101 10011110 10010100 01010000 10010001 10011110 10010100 01010000 10010010 10011111 10100111 01011100 01010000 10010010 10010101 10011110 10010100 01010000 10010001 10011110 10010100 01010000 10010010 10011111 10100111 01011100 00111010
01111001 10100010 10011111 10011110 01010000 10010001 10011110 10010100 01010000 10100011 10100100 10010101 10010101 10011100 01010000 10100111 10011001 10011100 10011100 01010000 10010010 10010101 10011110 10010100 01010000 10010001 10011110 10010100 01010000 10010010 10011111 10100111 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
01110010 10100101 10011001 10011100 10010100 01010000 10011001 10100100 01010000 10100101 10100000 01010000 10100111 10011001 10100100 10011000 01010000 10100011 10011001 10011100 10100110 10010101 10100010 01010000 10010001 10011110 10010100 01010000 10010111 10011111 10011100 10010100 01011100 00111010
10000011 10011001 10011100 10100110 10010101 10100010 01010000 10010001 10011110 10010100 01010000 10010111 10011111 10011100 10010100 01011100 01010000 10100011 10011001 10011100 10100110 10010101 10100010 01010000 10010001 10011110 10010100 01010000 10010111 10011111 10011100 10010100 01011100 00111010
01110010 10100101 10011001 10011100 10010100 01010000 10011001 10100100 01010000 10100101 10100000 01010000 10100111 10011001 10100100 10011000 01010000 10100011 10011001 10011100 10100110 10010101 10100010 01010000 10010001 10011110 10010100 01010000 10010111 10011111 10011100 10010100 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
10000011 10011001 10011100 10100110 10010101 10100010 01010000 10010001 10011110 10010100 01010000 10010111 10011111 10011100 10010100 01010000 10100111 10011001 10011100 10011100 01010000 10010010 10010101 01010000 10100011 10100100 10011111 10011100 10010101 10011110 01010000 10010001 10100111 10010001 10101001 01011100 00111010
10000011 10100100 10011111 10011100 10010101 10011110 01010000 10010001 10100111 10010001 10101001 01011100 01010000 10100011 10100100 10011111 10011100 10010101 10011110 01010000 10010001 10100111 10010001 10101001 01011100 00111010
10000011 10011001 10011100 10100110 10010101 10100010 01010000 10010001 10011110 10010100 01010000 10010111 10011111 10011100 10010100 01010000 10100111 10011001 10011100 10011100 01010000 10010010 10010101 01010000 10100011 10100100 10011111 10011100 10010101 10011110 01010000 10010001 10100111 10010001 10101001 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
10000011 10010101 10100100 01010000 10010001 01010000 10011101 10010001 10011110 01010000 10100100 10011111 01010000 10100111 10010001 10100100 10010011 10011000 01010000 10010001 10011100 10011100 01010000 10011110 10011001 10010111 10011000 01011100 00111010
10000111 10010001 10100100 10010011 10011000 01010000 10010001 10011100 10011100 01010000 10011110 10011001 10010111 10011000 10100100 01011100 01010000 10100111 10010001 10100100 10010011 10011000 01010000 10010001 10011100 10011100 01010000 10011110 10011001 10010111 10011000 10100100 01011100 00111010
10000011 10010101 10100100 01010000 10010001 01010000 10011101 10010001 10011110 01010000 10100100 10011111 01010000 10100111 10010001 10100100 10010011 10011000 01010000 10010001 10011100 10011100 01010000 10011110 10011001 10010111 10011000 10100100 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
10000011 10100101 10100000 10100000 10011111 10100011 10010101 01010000 10100100 10011000 10010101 01010000 10011101 10010001 10011110 01010000 10100011 10011000 10011111 10100101 10011100 10010100 01010000 10010110 10010001 10011100 10011100 01010000 10010001 10100011 10011100 10010101 10010101 10100000 01011100 00111010
01110110 10010001 10011100 10011100 01010000 10010001 10100011 10011100 10010101 10010101 10100000 01011100 01010000 10010110 10010001 10011100 10011100 01010000 10010001 10100011 10011100 10010101 10010101 10100000 01011100 00111010
10000011 10100101 10100000 10100000 10011111 10100011 10010101 01010000 10100100 10011000 10010101 01010000 10011101 10010001 10011110 01010000 10100011 10011000 10011111 10100101 10011100 10010100 01010000 10010110 10010001 10011100 10011100 01010000 10010001 10100011 10011100 10010101 10010101 10100000 01101111 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
01110111 10011001 10100110 10010101 01010000 10011000 10011001 10011101 01010000 10010001 01010000 10100000 10011001 10100000 10010101 01010000 10100100 10011111 01010000 10100011 10011101 10011111 10011011 10010101 01010000 10010001 10011100 10011100 01010000 10011110 10011001 10010111 10011000 10100100 01011100 00111010
10000011 10011101 10011111 10011011 10010101 01010000 10010001 10011100 10011100 01010000 10011110 10011001 10010111 10011000 10100100 01011100 01010000 10100011 10011101 10011111 10011011 10010101 01010000 10010001 10011100 10011100 01010000 10011110 10011001 10010111 10011000 10100100 01011100 00111010
01110111 10011001 10100110 10010101 01010000 10011000 10011001 10011101 01010000 10010001 01010000 10100000 10011001 10100000 10010101 01010000 10100100 10011111 01010000 10100011 10011101 10011111 10011011 10010101 01010000 10010001 10011100 10011100 01010000 10011110 10011001 10010111 10011000 10100100 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
11000101 >1 1111011010111101 >>1 10100 >1 100111111 >>1
11000101 >1 1111101111111101 >>1 10100 >1 100111111 >>1
11000101 >1 1111101111111101 >>1 11111 >1 11101111111111 >>1
00110000
11000101 >1 0000001111111100 >>1 00000
11000101 >1 1111101111111101 >>1 11111 >1 11111111111111 >>1
0000000
