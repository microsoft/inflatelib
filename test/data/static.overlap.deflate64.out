
repeat (144) {
    "London Bridge is falling down,\n"
    "Falling down, falling down,\n"
    "London Bridge is falling down,\n"
    "My fair Lady.\n"
    "\n"
    "Build it up with wood and clay,\n"
    "Wood and clay, wood and clay,\n"
    "Build it up with wood and clay,\n"
    "My fair Lady.\n"
    "\n"
    "Wood and clay will wash away,\n"
    "Wash away, wash away,\n"
    "Wood and clay will wash away,\n"
    "My fair Lady.\n"
    "\n"
    "Build it up with bricks and mortar,\n"
    "Bricks and mortar, bricks and mortar,\n"
    "Build it up with bricks and mortar,\n"
    "My fair Lady.\n"
    "\n"
    "Bricks and mortar will not stay,\n"
    "Will not stay, will not stay,\n"
    "Bricks and mortar will not stay,\n"
    "My fair Lady.\n"
    "\n"
    "Build it up with iron and steel,\n"
    "Iron and steel, iron and steel,\n"
    "Build it up with iron and steel,\n"
    "My fair Lady.\n"
    "\n"
    "Iron and steel will bend and bow,\n"
    "Bend and bow, bend and bow,\n"
    "Iron and steel will bend and bow,\n"
    "My fair Lady.\n"
    "\n"
    "Build it up with silver and gold,\n"
    "Silver and gold, silver and gold,\n"
    "Build it up with silver and gold,\n"
    "My fair Lady.\n"
    "\n"
    "Silver and gold will be stolen away,\n"
    "Stolen away, stolen away,\n"
    "Silver and gold will be stolen away,\n"
    "My fair Lady.\n"
    "\n"
    "Set a man to watch all nigh,\n"
    "Watch all night, watch all night,\n"
    "Set a man to watch all night,\n"
    "My fair Lady.\n"
    "\n"
    "Suppose the man should fall asleep,\n"
    "Fall asleep, fall asleep,\n"
    "Suppose the man should fall asleep?\n"
    "My fair Lady.\n"
    "\n"
    "Give him a pipe to smoke all night,\n"
    "Smoke all night, smoke all night,\n"
    "Give him a pipe to smoke all night,\n"
    "My fair Lady.\n"
}

repeat (1024) { 0 }

repeat (48) {
    "London Bridge is falling down,\n"
    "Falling down, falling down,\n"
    "London Bridge is falling down,\n"
    "My fair Lady.\n"
    "\n"
    "Build it up with wood and clay,\n"
    "Wood and clay, wood and clay,\n"
    "Build it up with wood and clay,\n"
    "My fair Lady.\n"
    "\n"
    "Wood and clay will wash away,\n"
    "Wash away, wash away,\n"
    "Wood and clay will wash away,\n"
    "My fair Lady.\n"
    "\n"
    "Build it up with bricks and mortar,\n"
    "Bricks and mortar, bricks and mortar,\n"
    "Build it up with bricks and mortar,\n"
    "My fair Lady.\n"
    "\n"
    "Bricks and mortar will not stay,\n"
    "Will not stay, will not stay,\n"
    "Bricks and mortar will not stay,\n"
    "My fair Lady.\n"
    "\n"
    "Build it up with iron and steel,\n"
    "Iron and steel, iron and steel,\n"
    "Build it up with iron and steel,\n"
    "My fair Lady.\n"
    "\n"
    "Iron and steel will bend and bow,\n"
    "Bend and bow, bend and bow,\n"
    "Iron and steel will bend and bow,\n"
    "My fair Lady.\n"
    "\n"
    "Build it up with silver and gold,\n"
    "Silver and gold, silver and gold,\n"
    "Build it up with silver and gold,\n"
    "My fair Lady.\n"
    "\n"
    "Silver and gold will be stolen away,\n"
    "Stolen away, stolen away,\n"
    "Silver and gold will be stolen away,\n"
    "My fair Lady.\n"
    "\n"
    "Set a man to watch all nigh,\n"
    "Watch all night, watch all night,\n"
    "Set a man to watch all night,\n"
    "My fair Lady.\n"
    "\n"
    "Suppose the man should fall asleep,\n"
    "Fall asleep, fall asleep,\n"
    "Suppose the man should fall asleep?\n"
    "My fair Lady.\n"
    "\n"
    "Give him a pipe to smoke all night,\n"
    "Smoke all night, smoke all night,\n"
    "Give him a pipe to smoke all night,\n"
    "My fair Lady.\n"
}
